main/src/ca/uwaterloo/flix/language/ast/TypedAst.scala

```diff
@@ -292,6 +293,16 @@ object TypedAst {
 
     case class FixpointProject(pred: Name.Pred, exp: TypedAst.Expression, tpe: Type, pur: Type, eff: Type, loc: SourceLocation) extends TypedAst.Expression
 
+    case class Error(m: CompilationMessage) extends TypedAst.Expression {
```

Comment:
I think this is OK for now but may change in the future.

Reply:
I had originally defined it as
```scala
case class Error(m: CompilationMessage, tpe: Type, pur: Type, eff: Type, loc: SourceLocation) extends TypedAst.Expression
```
but I wasn't sure what types to pass to the constructor when mapping from `KindedAst` to `TypedAst` i.e. at lines 2478-2479 in the Typer.

Reply:
Type.freshTypeVar and then of the right kind.

Reply:
And replace Type,  Type, Type by Type.Var, Type.Var, Type.Var